### PR TITLE
Alpha

### PR DESCRIPTION
**Pull request by loganwright**

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1,14 +1,16 @@
 import PackageDescription
 
+let alpha = Version(2,0,0, prereleaseIdentifiers: ["alpha"])
```
Comment (tanner0101) [inline Package.swift:3]: nice

File: Sources/VaporS3/Provider.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+extension Droplet {
+    /// Use this function to access the underlying
+    /// s3 object.
+    ///
+    /// make sure that VaporS3 has been added properly
+    /// before doing
+    public func s3() throws -> S3 {
+        guard let s3 = storage[s3StorageKey] as? S3 else { throw VaporS3Error.s3NotFound }
```
Comment (tanner0101) [inline Sources/VaporS3/Provider.swift]: I usually put something like `fooNotConfigured` or  `fooNotProperlyConfigured`. Something to let them know they haven't set up the provider correctly.

File: Sources/VaporS3/VaporS3Error.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    public var possibleCauses: [String] {
+        return [
+            "you're accessing drop.s3() before the provider has been added and booted properly"
+        ]
+    }
+
+    public var suggestedFixes: [String] {
+        return [
+            "make sure you're adding the s3 provider with `drop.addProvider`",
```
Comment (tanner0101) [inline Sources/VaporS3/VaporS3Error.swift:24]: ah the errors are pretty nice ...

**Review by tanner0101 — APPROVED** [review]

File: Sources/VaporS3/Provider.swift (hunk 2 above)
Comment (loganwright) [reply Sources/VaporS3/Provider.swift]: I'll give a quick rename

File: Sources/VaporS3/Provider.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import AWSSignatureV4
+import Vapor
+import S3
+
+private let s3StorageKey = "s3-provider:s3"
+
+public final class Provider: Vapor.Provider {
```
Comment (BrettRToomey) [inline Sources/VaporS3/Provider.swift:7]: 👍

**Review by BrettRToomey — COMMENTED** [review]
LGTM

**Review by BrettRToomey — APPROVED** [review]

**Pull request merged**